pkg/metrics/instance/batchapplyerror.go

```diff
@@ -0,0 +1,63 @@
+package instance
+
+import "fmt"
+
+// BatchApplyError contains the failed configs with error, successful configurations, and then if there is an error
+// that is not tied to a specific config then NonConfigError contains it
+type BatchApplyError struct {
+	Failed []BatchFailure
+	// NonConfigError is used when the error is not with a specific configuration, but likely the grouping
+	// or applying the group.
+	NonConfigError error
```

Comment:
This seems like a different error. I think BatchApplyError should only be responsible for indicating that the grouping was successful, but individual configs had issues. 

If at any point the grouping fails, we recover anyway, so returning both doesn't have a ton of value here.

Reply:
I think it would be simpler if we assume that the batch either fails or succeeds as a block. Any failure rolls back the configs.  Then we can return a standard error.

Reply:
I don't think we should do that. If a user happens to push a bad config through, it would prevent reshards from ever succeeding until that one config is resolved; it'd essentially be a noisy neighbor problem.